frontend/app/activeGame.js

```diff
@@ -31,7 +34,13 @@ export function onPlayerMove(playerMoveEvent) {
     const move = String(JSON.parse(playerMoveEvent.data));
     const items = move.split(',');
     // console.log(playerMoveEvent);
-    control.movePlayer(items[0], items[1], playerTokenInformation);
+
+    // had to assign variables to stop linter from complaining.
+    const player = items[0];
+    currentPlayer = player;
+    const endPosition = items[1];
+    playerPositions[currentPlayer].end = parseInt(endPosition, 10);
+    startAnimation();
```

Comment:
Does this code assume there will only be one player’s changed position in the playerMove event? The backend is generating a list of moves, so there could potentially be several.

Would it work to do the following instead?

```javascript
export function onPlayerMove(playerMoveEvent) {
    logEvents.logMoveEvent(playerMoveEvent);
    const moves = JSON.parse(playerMoveEvent.data);
    for (let i = 0; i < moves.length; i++) {
        currentPlayer = moves[i][0]
        playerPositions[currentPlayer].end  = moves[i][1];
        startAnimation();
    }
}
```

Also, the `playerMove` event currently includes both the position the player is moving to and the position the player is moving from – would using that info simplify things? (Event format: `[[<user id>, <new position>, <old position>], …]`.)

EDIT: Hmmm, now that I look at it a bit more (including how the animation works), handling multiple moves would take a bit more effort, because you’d have to figure out how to do multiple animations at once. For the moment it’s probably fine to assume there’ll only be one move event at any one time.

Reply:
@undergroundquizscene Ya it does assume there is only one player at a time changing positions. Which i think should be fine as its a turn based game. Once a player clicks end turn after the animation then everything works fine. If all players clicked roll-die and end turn immediately after each other then there is a chance of an error being logged to the console. But it still manages to work for the most part. I was gonna reduce the time for the set-interval to somewhat solve this problem and assume the player will wait. 

I didn't realise the playerMove event had the old positions, thanks for pointing that out. 
Ya i think i'll leave it as is if you don't mind :) It does it's job at least.

Reply:
Yeah, that’s a fair assumption alright. We probably should have written the SSE code with that assumption! Anyway, leaving it as it is sounds good :)